contrib/verifybinaries/verify.py

```diff
@@ -47,136 +172,441 @@ def parse_version_string(version_str):
     return version_base, version_rc, version_os
 
 
-def download_with_wget(remote_file, local_file=None):
-    if local_file:
-        wget_args = ['wget', '-O', local_file, remote_file]
-    else:
-        # use timestamping mechanism if local filename is not explicitly set
-        wget_args = ['wget', '-N', remote_file]
-
-    result = subprocess.run(wget_args,
+def download_with_wget(remote_file, local_file):
+    result = subprocess.run(['wget', '-O', local_file, remote_file],
                             stderr=subprocess.STDOUT, stdout=subprocess.PIPE)
     return result.returncode == 0, result.stdout.decode().rstrip()
 
 
+def download_lines_with_urllib(url) -> t.Tuple[bool, t.List[str]]:
+    """Get (success, text lines of a file) over HTTP."""
+    try:
+        return (True, [
+            line.strip().decode() for line in urllib.request.urlopen(url).readlines()])
+    except urllib.request.HTTPError as e:
+        log.warning(f"HTTP request to {url} failed (HTTPError): {e}")
+    except Exception as e:
+        log.warning(f"HTTP request to {url} failed ({e})")
+    return (False, [])
+
+
+def verify_with_gpg(
+    signature_filename,
+    output_filename: t.Optional[str] = None
+) -> t.Tuple[int, str]:
+    args = [
+        'gpg', '--yes', '--decrypt', '--verify-options', 'show-primary-uid-only',
+        '--output', output_filename if output_filename else '', signature_filename]
```

Comment:
GPG's manpage suggests using `--status-fd` (or `--status-file`) to for unattended usage of gpg. This outputs machine parseable lines to the specified file descriptor (or file) that look like:

```
[GNUPG:] NEWSIG fanquake@gmail.com
[GNUPG:] KEY_CONSIDERED E777299FC265DD04793070EB944D35F9AC3DB76A 0
[GNUPG:] SIG_ID IpS4EHolPjGh7he2GCBYNha53aY 2022-11-17 1668711593
[GNUPG:] KEY_CONSIDERED E777299FC265DD04793070EB944D35F9AC3DB76A 0
[GNUPG:] GOODSIG 2EEB9F5CC09526C1 Michael Ford (bitcoin-otc) <fanquake@gmail.com>
[GNUPG:] VALIDSIG CFB16E21C950F67FA95E558F2EEB9F5CC09526C1 2022-11-17 1668711593 0 4 0 1 8 00 E777299FC265DD04793070EB944D35F9AC3DB76A
[GNUPG:] KEY_CONSIDERED E777299FC265DD04793070EB944D35F9AC3DB76A 0
[GNUPG:] KEY_CONSIDERED E777299FC265DD04793070EB944D35F9AC3DB76A 0
[GNUPG:] TRUST_UNDEFINED 0 pgp
[GNUPG:] NEWSIG
[GNUPG:] KEY_CONSIDERED 6B002C6EA3F91B1B0DF0C9BC8F617F1200A6D25C 0
[GNUPG:] SIG_ID TVTce/SMh2FBof7NtPId5HC5MBo 2022-11-18 1668745859
[GNUPG:] KEY_CONSIDERED 6B002C6EA3F91B1B0DF0C9BC8F617F1200A6D25C 0
[GNUPG:] GOODSIG BA03F4DBE0C63FB4 Gloria Zhao <gloriazhao@berkeley.edu>
[GNUPG:] VALIDSIG F19F5FF2B0589EC341220045BA03F4DBE0C63FB4 2022-11-18 1668745859 0 4 0 1 8 00 6B002C6EA3F91B1B0DF0C9BC8F617F1200A6D25C
[GNUPG:] KEY_CONSIDERED 6B002C6EA3F91B1B0DF0C9BC8F617F1200A6D25C 0
[GNUPG:] KEY_CONSIDERED 6B002C6EA3F91B1B0DF0C9BC8F617F1200A6D25C 0
[GNUPG:] TRUST_UNDEFINED 0 pgp
```

This should make the parsing done in `parse_gpg_output` simpler as the output should be consistently formatted for machine parsing.

A description of this format can be found at https://git.gnupg.org/cgi-bin/gitweb.cgi?p=gnupg.git;a=blob;f=doc/DETAILS;h=e064c9d214506b4b9cea95448677afb2f885caee;hb=refs/heads/STABLE-BRANCH-2-2#l414

The docs also suggest using `--with-colons`, although it doesn't seem to make a difference in my testing.